src/scripts/Viewer.vue

```diff
@@ -1,36 +1,78 @@
 <template>
 	<div class="wrapper">
 		<div class="viewer">
-			<button class="viewer--close" @click="close"></button>
-			<div class="viewer--container" :style="'background-image:url(' + imgSrc+ ')'"></div>
+			<button class="viewer__close" @click="close()"></button>
+			<div class="swiper-container">
+				<div class="swiper-wrapper">
+					<div class="swiper-slide" v-for="(slide, index) in list" :key="index">
+						<img v-if="shouldRender(index)" class="viewer__media" :src="src(slide)" :alt="slide.name">
+					</div>
+				</div>
+			</div>
+			<button class="viewer__button--next">Next</button>
+			<button class="viewer__button--prev">Prev</button>
 		</div>
 	</div>
 </template>
 
 <script>
 const app = require('../../package.json');
 
+import Swiper from 'swiper';
+
 export default {
 	name: "Viewer",
+	data () {
+		return {
+			swiper : null
+		}
+	},
 	methods: {
 		close() {
 			this.$router.push('/');
+		},
+
+		src (item) {
+			let path = OC.joinPaths(
+				OC.getRootPath(),
+				'remote.php/dav/files',
+				OC.getCurrentUser().uid,
```

Comment:
this will clash later once we handle public link shares. can you raise a ticket for public link share support ? (will likely require extra effort)